src/bytearray_stubs.c

```diff
@@ -3,14 +3,15 @@
 
 #include <string.h>
 
+#include "caml/mlvalues.h"
 #include "caml/intext.h"
 #include "caml/bigarray.h"
 #include "caml/memory.h"
 
 CAMLprim value ml_marshal_to_bigarray(value v, value flags)
 {
   char *buf;
-  long len;
```

Comment:
I am not really following this.  Certainly it is wrong to use "long" as the size of long is not fixed.    On i386, the normal (unix) convention is ILP32, and on x86_64 LP64.   So with the use of long, I'd expect far more trouble than we are seeing.   Can you add to the commit message an explanation of how this is wrong, and why you believe that after this change we will have no wire protocol break compared to all systems except windows-x86_64-mingw?

intnat is basically a typedef for value, but I can't find that.

In config.h I see a lack of ILP64, like Alpha.  But that's an ocaml bug, not a unison bug, I think.

Or is this about the bits ending up in different places in &len and happy aliasing depending on LE vs BE?

Were there warnings before?

Reply:
I don't follow you. Maybe things are clearer knowing that long is 32-bit even on Windows x86_64?

> Were there warnings before?

Yes: look at Win64 CI/CD build logs (for example https://github.com/bcpierce00/unison/runs/1311424481?check_suite_focus=true ). With @tleedjarv 's patch, the warning disappears.

Reply:
So Windows x86_64 is IL32 P64?

What I don't understand is that long changes size from i386 to x86_64 on unix, so if this sizeof(long) issue caused trouble on Windows (which I believe from the reports of course), then I don't understand why it has caused no other trouble.

Reply:
So is it that intnat is defined to be the natural word size, which is long=32 on i386, long=64 on x86-64, and ends up long long=64 on windows x86-64 mingw?   In other words, long and intnat are both variable, but one tracks what you'd expect and long is strange on Windows x86-64 mingw?

Where is intnat defined?  I grepped the ocaml .h files and didn't find it, but I admit to not reading them all.

Reply:
Yes to all.

> Where is intnat defined?

In config.h.